ehrapy/tools/causal/_dowhy.py

```diff
@@ -0,0 +1,362 @@
+        "backdoor.econml.metalearners.XLearner",
+    ],
+    refute_methods: None
+    | list[str]
+    | (
+        list[
+            Literal[
+                "placebo_treatment_refuter", "random_common_cause", "data_subset_refuter", "add_unobserved_common_cause"
+            ]
+        ]
+    ) = None,
+    print_causal_estimate: bool = False,
+    print_summary: bool = True,
+    return_as: Literal["estimate", "refute", "estimate+refute"] = "estimate",
+    show_graph: bool = False,
+    show_refute_plots: bool | Literal["colormesh", "contour", "line"] | None = None,
+    attempts: int = 10,
+    *,
+    identify_kwargs: dict[str, Any] | None = None,
+    estimate_kwargs: dict[str, Any] | None = None,
+    refute_kwargs: dict[str, Any] | None = None,
+) -> tuple[dowhy.CausalEstimate, dict[str, str | dict[str, float]]]:
+    """
+    Performs causal inference on an AnnData object using the specified causal model and returns a tuple containing the causal estimate and the results of any refutation tests.
+
+    Args:
+        adata: An AnnData object containing the input data.
+        graph: A str representing the causal graph to use.
+        treatment: A str representing the treatment variable in the causal graph.
+        outcome: A str representing the outcome variable in the causal graph.
+        estimation_method: An optional Literal specifying the estimation method to use. Defaults to "backdoor.propensity_score_stratification".
+        refute_methods: An optional List of Literal specifying the methods to use for refutation tests. Defaults to ["placebo_treatment_refuter", "random_common_cause", "data_subset_refuter"].
+        print_causal_estimate: Whether to print the causal estimate or not, default is False.
+        print_summary: Whether to print the causal model summary or not, default is True.
+        return_as: An optional Literal specifying the type of output to return. Defaults to "summary".
+        show_graph: Whether to display the graph or not, default is False.
+        show_refute_plots: Whether to display the refutation plots or not, default is False.
+        attempts: Number of attempts to try to generate a valid causal estimate, default is 10.
+        identify_kwargs: Optional keyword arguments for dowhy.CausalModel.identify_effect().
+        estimate_kwargs: Optional keyword arguments for dowhy.CausalModel.estimate_effect().
+        refute_kwargs: Optional keyword arguments for dowhy.CausalModel.refute_estimate().
+
+    Returns:
+        A tuple containing the causal estimate and a dictionary of the results of any refutation tests.
+
+    Raises:
+        TypeError: If adata, graph, treatment, outcome, refute_methods, estimation_method, or return_as is not of the expected type.
+        ValueError: If refute_methods or estimation_method contains an unknown value, or if return_as is an unknown value.
+
+    Examples:
+        >>> data = dowhy.datasets.linear_dataset(
```

Comment:
Guess I should add this to the ehrapy datasets at some point to make it easier

Reply:
For testing purposes? Through the causal module we'll have dowhy as a dependency anyway, no? Might as well use their function in the test, no?